src/app/content/components/Page/PageComponent.tsx

```diff
@@ -43,13 +42,20 @@ export default class PageComponent extends Component<PagePropTypes> {
     return cleanContent.body.innerHTML;
   };
 
-  public getCurrentScrollTarget = (prevProps: PagePropTypes) => {
+  public getScrollTargets = (prevProps: PagePropTypes): ScrollTargets => {
```

Comment:
i don't think we should be adding logic to this component to handle scrolling, or identifying scroll targets. i can understand why the centralized one in navigation might not have worked, since each target type has different data they need. i also liked it better when a scroll target had one identifier and a type, rather than one object with a bunch of keys that need to be known everywhere.

since i guess its possible to have both a search result and a hash target, it seems like we need a way to somehow generate a list of the current scroll targets. i don't think we should have a bunch of logic in pageComponent for that, maybe if navigation also isn't a good place we can put something in each manager which will return either null or a scrolltarget.

and in the scrolltargetmanager we need to be looking for targets which were not in the previous prop, so that we only catch it when its added, otherwise we'll be scrolling to it every time we re-render.